### DISCUSSION TITLE
Calculation of specific heat, renormalized harmonic FCs and alamode version 1.3.0 query

### DISCUSSION THREAD

**Discussion by apandit2021** [ttadano/alamode#70] [Q&A]
Dear Dr. Tadano,

It would be a great help if you could respond to my following questions regarding the use of alamode:
1) I can calculate specific heat capacity (C_v) as a function of temperature using alamode. But is it also possible to evaluate C_v as a function of the phonon frequency at a certain temperature? I see C_v is dependent on the frequency as well in its formulation.
2) I know the process but, using tools/dfc2, how does the alamode generate/calculate renormalized harmonic FCs with anharmonic correction in terms of equation/formula? 
3) I see the latest alamode version 1.3.0 available. Could you please let me know what specific features are added/changed in it as compared to the previous version?

Best,
Abhiyan

**Comment by ttadano** [comment]
1. Is C_v here the mode-specific heat? If so, C_v(omega) = \hbar\omega (\partial n)/(\partial T), where n is the Bose-Einstein distribution function. This can be calculated analytically.
2. `tools/dfc2` just computes \Phi_ij^{harmonic} + \Delta \Phi_{ij}^{SCP}. The first term is the harmonic force constants in real space and the second one is the anharmonic correction obtained from an SCP calculation. \Delta \Phi_{ij}^{SCP} is also in real space, which is computed from the correction to the dynamical matrix by inverse Fourier transform.
3. Please see https://github.com/ttadano/alamode/releases/tag/v.1.3.0

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

Thank you for the clarification. In above part answer 1, could you please let me know what is meant by partial n and partial T?

Best,
Abhiyan

**Reply by ttadano** [reply]
It's a partial derivative of n by temperature. \partial is a LaTeX command: https://www.tutorialspoint.com/tex_commands/partial.htm

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

Okay I got that now. But how can we do the partial derivative with respect to temperature (T) if I want to calculate C_v as function of frequency at a finite T, let's say at 300 K?

Best,
Abhiyan

**Reply by ttadano** [reply]
The definition of n is <img src="https://render.githubusercontent.com/render/math?math=n(\omega) =[e^{\hbar\omega/kT}-1]^{-1}">. Its partial derivative by <img src="https://render.githubusercontent.com/render/math?math=T"> is quite straightforward.

**Reply by apandit2021** [reply]
Dear Dr. Tadano,

Oh, I see. I thought del_T means the temperature change. But here we will operate partial derivative within n and then input the finite T.
Thank you so much for your prompt and informative response.

Best,
Abhiyan
